Security/Authentication/Token/LtiLaunchRequestToken.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Bundle\Lti1p3Bundle\Security\Authentication\Token;
+
+use OAT\Library\Lti1p3Core\Launch\Validator\LtiLaunchRequestValidationResult;
+use OAT\Library\Lti1p3Core\Message\LtiMessageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\AbstractToken;
+
+class LtiLaunchRequestToken extends AbstractToken
+{
+    /** @var string[] */
+    private $roleNames = [];
+
+    /** @var LtiLaunchRequestValidationResult|null */
+    private $validationResult;
+
+    public function __construct(LtiLaunchRequestValidationResult $validationResult = null)
+    {
+        parent::__construct($this->roleNames);
```

Comment:
I think you can avoid to provide this parameter which is always empty. 
Even if it wouldn't, we don't use the parent roleNames :smiley:

Reply:
it's to match the TokenInterface of the SF security component.

Since I do not want to implement this (very verbose) interface, I use the abstraction given in AbstractToken, which implies to pass a role array at construction.

Reply:
I understand, what I mean is that $this->roleNames will be always empty here right ? then you could just call
 parent::__construct();

Reply:
as discussed together, fixed